src/common/config.cc

```diff
@@ -474,6 +475,16 @@ void md_config_t::parse_env(ConfigValues& values,
       _set_val(values, tracker, dir, *o, CONF_ENV, &err);
     }
   }
+  const char *pod_req = getenv("POD_MEMORY_REQUEST");
+  if (pod_req) {
```

Comment:
Note that `POD_MEMORY_REQUEST` will be zero if not specified in the pod definition. During pod's scheduling, if `memory.request` is not set it equals `memory.limit`, however, that's not reflected in pod's properties when running.

Reply:
yeah, but (1) we don't want to set osd_memory_target to the actual limit or we'll quickly crash, and (2) we pick up that limit * the ratio from the cgroup directly in the next patch

Reply:
oh, you mean we shoudl do something different if the value is zero?

Reply:
Yeah, if value is `0` then we switch to reading `/sys`. So I guess the condition should be `if (pod_req) && (pod_req != 0)`.

Reply:
Is it possible that the check for `pod_req != 0` was never implemented? I am seeing some strange behavior where the memory target is set to 100G.

Reply:
@salanki I had the same concern, I checked with @liewegas and it seems that if the value is 0 then it's treated as if it isn't set at all. So we are covered.